applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -398,6 +415,48 @@ static struct bt_audio_stream_ops stream_ops = {
 	.recv = stream_recv_cb,
 };
 
+static void work_stream_start(struct k_work *work)
+{
+	int ret;
+	struct worker_data work_data;
+
+	ret = k_msgq_get(&kwork_msgq, &work_data, K_NO_WAIT);
+	if (ret) {
+		LOG_ERR("Cannot get info for start stream");
+	}
+
+	if (work_data.dir == BT_AUDIO_DIR_SINK) {
+		ret = bt_audio_stream_start(&headsets[work_data.channel_index].sink_stream);
+	} else if (work_data.dir == BT_AUDIO_DIR_SOURCE) {
+		ret = bt_audio_stream_start(&headsets[work_data.channel_index].source_stream);
+	}
+
+	work_data.retries++;
+
+	if (ret == -EBUSY) {
+		if (work_data.retries < CIS_CONN_RETRY_TIMES) {
+			LOG_WRN("Got connect error from stream %d Retrying. code: %d count: %d",
+				work_data.channel_index, ret, work_data.retries);
+			ret = k_msgq_put(&kwork_msgq, &work_data, K_NO_WAIT);
+			if (ret) {
+				LOG_ERR("No space in the queue for the bond");
+			}
+			/* Delay added to prevent controller overloading */
+			k_work_reschedule(&headsets[work_data.channel_index].stream_start_work,
+					  K_MSEC(300));
+		} else {
+			LOG_ERR("Could not connect ch %d after %d retries", work_data.channel_index,
+				work_data.retries);
+			bt_conn_disconnect(headsets[work_data.channel_index].headset_conn,
+					   BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+		}
+	} else if (ret != 0) {
+		LOG_ERR("Failed to establish CIS, ret = %d", ret);
+		bt_conn_disconnect(headsets[work_data.channel_index].headset_conn,
+				   BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
```

Comment:
```suggestion
	if ((ret == -EBUSY) && (work_data.retries < CIS_CONN_RETRY_TIMES)) {
		LOG_WRN("Got connect error from stream %d Retrying. code: %d count: %d",
			work_data.channel_index, ret, work_data.retries);
		ret = k_msgq_put(&kwork_msgq, &work_data, K_NO_WAIT);
		if (ret) {
			LOG_ERR("No space in the queue for the bond");
		}
		/* Delay added to prevent controller overloading */
		k_work_reschedule(&headsets[work_data.channel_index].stream_start_work,
				  K_MSEC(300));
	} else {
		LOG_ERR("Could not connect ch %d after %d retries, ret: %d", work_data.channel_index,
			work_data.retries, ret);
		bt_conn_disconnect(headsets[work_data.channel_index].headset_conn,
				   BT_HCI_ERR_REMOTE_USER_TERM_CONN);
	}
```